Meter.c

```diff
@@ -222,11 +261,21 @@ static void BarMeterMode_draw(Meter* this, int x, int y, int w) {
 
    // First draw in the bar[] buffer...
    int offset = 0;
+   double total = this->total;
+   if (total <= 0.0) {
+      // Update "total" value for dynamic scale.
+      // "total" is only updated in Bar meter mode.
+      double sum = Meter_computeSum(this);
+      if (sum > -(this->total)) {
+         this->total = -sum;
+      }
+      total = -(this->total);
+   }
    for (uint8_t i = 0; i < this->curItems; i++) {
       double value = this->values[i];
-      value = CLAMP(value, 0.0, this->total);
+      value = CLAMP(value, 0.0, total);
       if (value > 0) {
-         blockSizes[i] = ceil((value / this->total) * w);
+         blockSizes[i] = ceil((value / total) * w);
```

Comment:
I'll keep this pair of parentheses. (It clearly labels that division goes first. When it comes to multiplying a fraction, we usually do multiplication before division, but this is division first to avoid an overflow.)